server/src/test/java/org/apache/druid/discovery/BaseNodeRoleWatcherTest.java

```diff
@@ -116,6 +116,19 @@ public void testGeneralUseSimulation()
     assertListener(listener3, true, nodesAdded, nodesRemoved);
   }
 
+  @Test(timeout = 60_000L)
+  public void testTimeOutAfterInitialization() throws InterruptedException
```

Comment:
Please also include tests for the case where some nodes are added, but initialization is never called. In the test cases, verify the behavior of these scenarios:

- `registerListener` is called before the timeout: verify the listener does get the proper notifications
- `getAllNodes` is called before the timeout: verify the call does return the list of nodes

These scenarios should be verified in separate test cases to ensure that they each work in isolation.

Reply:
Added tests